lib/administrate/namespacing_helpers.rb

```diff
@@ -0,0 +1,65 @@
+module Administrate
+  module NamespacingHelpers
```

Comment:
I think there isn't value in putting these in a mixin, as this code is only referenced from one place: `RoutesGenerator`. If I'm reading the generator, then I find myself wondering "where is generate_resource_routes? Oh, it's in a mixin, let's go there". In fact there's one additional jump, as my search for this code started at the template `routes.rb.erb`. If we were trying to be DRY it could be worth it, but in our case there's less reason for it.

Also, of all these methods, the only one that is used by other code is `generate_resource_routes`. The rest are only used internally and should be private, so that the reader doesn't have to wonder what they are for.

Following on what I said at https://github.com/thoughtbot/administrate/pull/1812/files#r540295544 about factoring code out into a class: that would be a good reason to separate the code. We could this complex, recursive code somewhere that it can be unit-tested, so that we can make sure that every case is covered without having to make our example app any more complex.

Reply:
I'll look at this more on the weekend - but if I understand the suggestion is to replace the module with a class allowing just he `generate_resource_routes` to be public?

Reply:
Not just these methods, but rather all the code that takes part in this complicated calculation of the routes (I'm commenting in passing now, so I'm not sure what else is involved). I linked the wrong comment earlier, I meant https://github.com/thoughtbot/administrate/pull/1812#discussion_r540294128, which includes an example.

Reply:
I am struggling to see how it works as a class - probably as its part of a larger issue.. As we would want to call a function of that class, `generate_resource_routes`, from a erb file generated by the routes_generator... (is this possible?)